### DISCUSSION TITLE
How does Elsa 3.1.3 implement the If activity step in the designer?

### DISCUSSION THREAD

**Discussion by shxylgithub** [elsa-workflows/elsa-core#5622] [General]
How does Elsa 3.1.3 implement the If activity step in the designer?

**Comment by glime-ai** [comment]
In Elsa 3.1.3, the `If` activity is implemented as a custom activity that models a simple `if` statement. The `If` activity has three properties: `Condition`, `Then`, and `Else`. The `Condition` property is of type `Input<bool>` and determines which branch of the `If` activity will be executed. The `Then` and `Else` properties are of type `IActivity` and represent the activities to execute if the condition is true or false, respectively.

When the `If` activity is executed, it evaluates the `Condition` and then schedules either the `Then` or `Else` activity for execution based on the result of the condition. After the child activity completes, a callback method `OnChildCompleted` is invoked to complete the `If` activity.

To use the `If` activity in a workflow, you would assign it to the `Root` property of the workflow, as it implements the `IActivity` interface. If you need to run multiple activities, you can use a container activity such as `Sequence` or `Flowchart` as the root activity.

Here is an example of how to use the `If` activity in a workflow:

```csharp
var workflow = new Workflow
{
    Root = new If
    {
        Condition = new(true),
        Then = new WriteLine("Hello World!"),
        Else = new WriteLine("Goodbye cruel world!")
    }
};
```

In the Elsa Designer, you would visually add the `If` activity to your workflow and configure its properties accordingly. However, the provided context does not include specific details about the designer interface for Elsa 3.1.3, so I cannot provide a step-by-step guide on how to implement the `If` activity in the designer without additional information.

If you need further assistance or clarification on how to use the `If` activity in the Elsa Designer, please let me know!

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
